src/main/java/io/seqera/tower/cli/commands/runs/DumpCmd.java

```diff
@@ -167,15 +171,47 @@ private void dumpTasks(PrintWriter progress, TarArchiveOutputStream out, Long ws
     private void dumpWorkflowDetails(PrintWriter progress, TarArchiveOutputStream out, Long wspId) throws ApiException, IOException {
         progress.println(ansi("- Workflow details"));
 
+        // General workflow info
         DescribeWorkflowResponse workflowResponse = workflowById(wspId, id);
         Workflow workflow = workflowResponse.getWorkflow();
         if (workflow == null) {
             throw new TowerException("Unknown workflow");
         }
+
+        // Launch info
+        Launch launch = null;
+        Long pipelineId = null;
+        if (workflow.getLaunchId() != null) {
+
+            launch = launchById(wspId, workflow.getLaunchId());
+
+            DescribeWorkflowLaunchResponse wfLaunchResponse = workflowLaunchById(wspId, workflow.getId());
+            if (wfLaunchResponse != null && wfLaunchResponse.getLaunch() != null) {
+                pipelineId = wfLaunchResponse.getLaunch().getPipelineId();
+            }
+        }
+
+        // User info
+        Long userId = workflow.getOwnerId();
+        String userMail = null;
+        String userName = workflow.getUserName();
+        try {
```

Comment:
Same here. This Try catch could be hidden on a separate method with a descriptive name. That way all this method becomes way simpler and easy to understand.

Reply:
I think that moving fragments of the code to a separate method makes sense if said method is getting reused more than once or if the fragment is overly complex. Otherwise we are just scattering the code around without actually reducing complexity.
I'll add some comments given that this filtering fragment is a bit convoluted. If we find an easier approach I'll update it.

Reply:
Comments are not pollution in the code base. We strongly encourage adding comments wherever is necessary as long as they explain the "why" rather than the "what" in the code.

> Which one looks cleaner?

This is subjective, sure the wrapper method is 3 lines but then you have moved the same code into two methods, so you end with N+3 lines and additional jumps that makes debugging more difficult.

P.S: Thanks a lot for the recommendation, I enjoyed reading that book back in the day. Let me return the favor and recommend you another [book](https://www.amazon.com/Pragmatic-Programmer-Journeyman-Master/dp/020161622X).
If you are more curious about this and other pragmatic principles please let me know.

Reply:
> moved the same code into two methods

Not really, what happens is that both methods will call a single method where all the duplicate code is. However, that part is not shown in the screenshot.

> makes debugging more difficult
How so?

Thanks for the book recommendation 👍🏻